File: tableau/TableauCoffre.java
package tableau;

import coffre.Coffre;
import deplacement.Deplacement;

public class TableauCoffre extends Tableau
{
private Coffre coffre;
	
	public TableauCoffre(int X, int Y, Coffre coffre)
	{
		super(X, Y);
		this.coffre = coffre;
	}

	public Coffre getCoffre() {
		return coffre;
	}
	public void setCoffre(Coffre coffre) {
		this.coffre = coffre;
	}
	
	public void evenement()
	{
		Coffre.ouverture(this);
		Deplacement.deplacement(this);
	}
}

File: Cs/tableau/TableauCoffre.cs
using Coffre;  
using Deplacement;

namespace TableauNamespace
{
    public class TableauCoffre : Tableau
    {
        private Coffre coffre;

        public TableauCoffre(int x, int y, Coffre coffre) : base(x, y)
        {
            this.coffre = coffre;
        }

        public Coffre GetCoffre()
        {
            return coffre;
        }

        public void SetCoffre(Coffre coffre)
        {
            this.coffre = coffre;
        }

        public override void Evenement()
        {
            Coffre.Ouverture(this);
            Deplacement.Deplacement(this);
        }
    }
}
